src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,214 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+import { autobind } from '@shopify/javascript-utilities/decorators';
+import { createUniqueIDFactory } from '@shopify/javascript-utilities/other';
+import { findFirstFocusableNode } from '@shopify/javascript-utilities/focus';
+import { addEventListener, removeEventListener } from '@shopify/javascript-utilities/events';
+
+import DropdownItem from './DropdownItem';
+import { classNames } from '@shopify/react-utilities/styles';
+import { DROPDOWN } from '../ThemeIdentifiers';
+import * as baseTheme from './Dropdown.scss';
+import { Keys } from '../../types';
+import Button from '../Button';
+
+export type Direction = 'up' | 'down' | 'left' | 'right';
+
+export interface DropdownItemProps {
+  children?: React.ReactNode;
+  disabled?: boolean;
+  divider?: boolean;
+  header?: boolean;
+  onClick?(): void;
+}
+
+export interface Props {
```

Comment:
@shyamk9 There is no onClick or onSelect prop here, how user will know what item he has clicked & how can they decide what operation they want to perform onclick of an item, like:
1. if they want to navigate to specific route
2. Use dropdown like a filter or action menu, something similar to the dropdowns target process have